crml/nft/src/benchmarking.rs

```diff
@@ -0,0 +1,274 @@
+*/
+
+//! NFT benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use super::*;
+
+use frame_benchmarking::{account, benchmarks, whitelisted_caller};
+use frame_system::RawOrigin;
+use sp_runtime::Permill;
+
+use crate::types::MAX_ENTITLEMENTS;
+use crate::{Module as Nft, MAX_COLLECTION_ID_LENGTH};
+
+// Create a collection for benchmarking
+// Returns the collection id, schema, and royalties schedule
+fn setup_collection<T: Trait>(creator: T::AccountId) -> (CollectionId, NFTSchema, RoyaltiesSchedule<T::AccountId>) {
+	// Collection id with max length
+	let collection_id = [1_u8; MAX_COLLECTION_ID_LENGTH as usize].to_vec();
+
+	// Schema with max. attributes
+	let schema = (0..MAX_SCHEMA_FIELDS)
+		.map(|i| {
+			let mut v = b"probably-a-decent-upper-length-for-an-attribte-name".to_vec();
+			v.push(i as u8);
+			(v, NFTAttributeValue::String(Default::default()).type_id())
+		})
+		.collect::<NFTSchema>();
+
+	// Royalties with max. entitled addresses
+	let royalties = RoyaltiesSchedule::<T::AccountId> {
+		entitlements: (0..MAX_ENTITLEMENTS)
+			.map(|_| (creator.clone(), Permill::from_percent(1)))
+			.collect::<Vec<(T::AccountId, Permill)>>(),
+	};
+
+	return (collection_id, schema, royalties);
+}
+
+// Create a token for benchmarking
+fn setup_token<T: Trait>(owner: T::AccountId) -> CollectionId {
+	let creator: T::AccountId = whitelisted_caller();
+	let (collection_id, schema, royalties) = setup_collection::<T>(creator.clone());
+	let _ = <Nft<T>>::create_collection(
+		RawOrigin::Signed(creator.clone()).into(),
+		collection_id.clone(),
+		schema.clone(),
+		Some(royalties.clone()),
+	)
+	.expect("created collection");
+	// all attributes max. length
+	let attributes = schema
+		.iter()
+		.map(|_| NFTAttributeValue::String([1_u8; 140_usize].to_vec()))
+		.collect::<Vec<NFTAttributeValue>>();
+	let _ = <Nft<T>>::create_token(
+		RawOrigin::Signed(creator.clone()).into(),
+		collection_id.clone(),
+		owner.clone(),
+		attributes.clone(),
+		Some(royalties),
+	)
+	.expect("created token");
+
+	return collection_id;
+}
+
+benchmarks! {
+	_{}
+
+	create_collection {
+		let creator: T::AccountId = account("creator", 0, 0);
+		let (collection_id, schema, royalties) = setup_collection::<T>(creator.clone());
+
+	}: _(RawOrigin::Signed(creator.clone()), collection_id.clone(), schema, Some(royalties))
+	verify {
+		assert_eq!(<Nft<T>>::collection_owner(&collection_id), Some(creator));
+	}
+
+	create_token {
+		let creator: T::AccountId = whitelisted_caller();
+		let owner: T::AccountId = account("owner", 0, 0);
+
+		let (collection_id, schema, royalties) = setup_collection::<T>(creator.clone());
+		let _ = <Nft<T>>::create_collection(RawOrigin::Signed(creator.clone()).into(), collection_id.clone(), schema.clone(), Some(royalties.clone())).expect("created collection");
+		// all attributes max. length
+		let attributes = schema.iter().map(|_| NFTAttributeValue::String([1_u8; 140_usize].to_vec())).collect::<Vec<NFTAttributeValue>>();
```

Comment:
Could you already make schema like this in setup_collection?

Reply:
If 140 is a maximum len that should be respected in the module, could you please add 140 as a defined const in the module and "use" it here?

Reply:
needs a const here, added an assert as a workaround